### PR TITLE
Changed Jenkinsfile to publish master branch to npmjs.org

### PR DESCRIPTION
**Pull request by ChrisPark89**
Issue #189 #266 

I removed 'Bump Version' stage because now we manually the change version in package.json when we release a new version.
However, we still change the version of `dev` branch and deploy it to internal artifactory whenever there is a new commit.

The deployment step will be skipped, 
A. if there isn't any new commit for both `dev` and `master` branch
B. if we don't change the version in package.json in `master` branch (This is for making documentation changes easy without releasing a new version)

### CODE REVIEW

**Comment by matthewpwilson** [comment]
I'd rather not merge this until we have approval to publish to npmjs.org to avoid doing it by mistake!

File: Jenkinsfile
Comment (matthewpwilson) [inline Jenkinsfile]: Suggest removing all the stuff relating `ci_skip` as we don't need this

File: Jenkinsfile
Comment (matthewpwilson) [inline Jenkinsfile]: I think the e-mail address will be needed in `.npmrc` as well

File: Jenkinsfile
Comment (matthewpwilson) [inline Jenkinsfile]: Is it possible to add `--registry $NPM_REGISTRY` to the `npm view` command to make sure we're not picking up anything different from `.npmrc`?

File: package.json
Comment (matthewpwilson) [inline package.json]: I think we should leave this at 0.5 for now so we can do a test publish

**Review by matthewpwilson — CHANGES_REQUESTED** [review]

File: Jenkinsfile
Comment (ChrisPark89) [reply Jenkinsfile]: done

File: package.json
Comment (ChrisPark89) [reply package.json]: done

File: Jenkinsfile
Comment (ChrisPark89) [reply Jenkinsfile]: which email address should I use?

File: Jenkinsfile
Comment (ChrisPark89) [reply Jenkinsfile]: done

File: package.json
Comment (matthewpwilson) [reply package.json]: Actually.... do we need to update this to 0.6 in order to do the test publish, sorry hadn't thought that through!

File: Jenkinsfile
Comment (matthewpwilson) [reply Jenkinsfile]: The credential is set up in Jenkins as a username and password so I think you should be able to get the email address with `$NPM_CREDENTIALS_USR` and the password with `$NPM_CREDENTIALS_PWD` as described here: https://jenkins.io/doc/book/pipeline/jenkinsfile/#usernames-and-passwords

Comment (ChrisPark89) [reply Jenkinsfile]: In fact email address is not needed as the token has the email information, however I changed NPM_CREDENTIALS to NPM_CREDENTIALS_PSW as this is the correct token.

**Review by matthewpwilson — APPROVED** [review]

**Pull request merged**
